### PR TITLE
Fix file descriptor processing

### PR DESCRIPTION
**Pull request by DemiMarie**
This adds assertions, strengthens input validation, and overall hardens the code.  I need to do more testing on the changes to `move_fd`.

### CODE REVIEW

**Comment by DemiMarie** [comment]
PipelineRetry

**Comment by marmarek** [comment]
```
Traceback (most recent call last):
  File "/usr/lib/python3.8/site-packages/splitgpg/tests.py", line 371, in test_080_option_parser
    splitgpg_needs_arg = self._check_if_options_takes_argument(
  File "/usr/lib/python3.8/site-packages/splitgpg/tests.py", line 352, in _check_if_options_takes_argument
    self.fail(
AssertionError: qubes-gpg-client --status-fd have not complained about --garbage options: Invalid FD argument.  Only decimal numbers are supported
```

**Comment by DemiMarie** [comment]
@marmarek oops, should be fixed

**Comment by DemiMarie** [comment]
PipelineRetry

File: src/multiplex.c [hunk 1]
```diff
@@ -35,33 +35,49 @@
...
     sa.sa_handler = sigchld_handler;
     sa.sa_flags = 0;
     sigemptyset(&sa.sa_mask);
-    sigaction(SIGCHLD, &sa, NULL);
+    if (sigaction(SIGCHLD, &sa, NULL) != 0) {
+        perror("sigaction");
+        exit(1);
+    }
+
+    sa.sa_handler = sigpipe_handler;
```
Comment (marmarek) [inline src/multiplex.c:73]:
```suggestion
    sa.sa_handler = SIG_IGN;
```
? but then, you'd need to reset it before exec...

File: src/multiplex.c [hunk 2]
```diff
@@ -79,7 +95,7 @@ void *process_in(struct thread_args *args) {
             switch (read_len) {
                 case 0:
                     fprintf(stderr, "EOF\n");
-                    exit(EXIT_SUCCESS);
+                    exit(EXIT_FAILURE);
```
Comment (marmarek) [inline src/multiplex.c:98]: Why? client -> daemon EOF should be normal, I think something like `qubes-gpg-client -k | head` may produce such result (exit early, before retrieving all the data).

**Pull request merged**